source/geod24/LocalRest.d

```diff
@@ -40,6 +40,16 @@
     methods, for some defined time or until unblocked.
     See `sleep`, `filter`, and `clearFilter` for more details.
 
+    Shutdown:
+    The control interface has a shutdown method. It may be called when one
+    wants to shut down the node and call its destructor. Additionally some
+    classes may wish to do resource deallocation separately from their
+    destructors, in particular when those methods themselves allocate
+    (D destructors may not allocate GC memory). Therefore, you may provide
```

Comment:
That is actually not a concern for localrest. The node is allocated using `scope`, which means it's stack allocated, which means the destructor runs deterministically and can allocate.

Reply:
Ah well it was just an example, I could remove it. Anyway how is a destructor supposed to know if the object it's cleaning is allocated on the heap vs the stack?

https://github.com/biozic/d2sqlite3/blob/0280ef1769f0e0907980a241b9ac68e444a1fbd5/source/d2sqlite3/database.d#L89

Reply:
Looking at this method, I thought: "Please don't implement this by catching `InvalidMemoryOperationError`... PLEASE".
[Welp](https://github.com/biozic/d2sqlite3/blob/08e77f5f2da3ead57d62c80226bd85e41c27fbd7/source/d2sqlite3/internal/memory.d#L70)